projects/graph/src/graph.py

```diff
@@ -2,8 +2,83 @@
 Simple graph implementation compatible with BokehGraph class.
 """
 
+class Queue:
+	def __init__(self):
+		self.size = 0
+		self.storage = []
+ 	def enqueue(self, item):
+		self.storage.append(item)
+		self.size += 1
+ 	def dequeue(self):
+		if self.size:
+			self.size -= 1
+		return self.storage.pop(0)
+ 	def len(self):
+		return self.size
 
+class Stack:
+    def __init__(self):
+        self.stack = []
+    def push(self, value):
+        self.stack.append(value)
+    def pop(self):
+        if self.size() > 0:
+            return self.stack.pop()
+        else:
+            return None
+    def size(self):
+        return len(self.stack)
+        
 class Graph:
     """Represent a graph as a dictionary of vertices mapping labels to edges."""
     def __init__(self):
-        pass  # TODO
+        self.vertices={}
+    def add_vertex(self,vertex_id):
+        self.vertices[vertex_id]=Vertex(vertex_id)
+    def add_edge(self,v1,v2):
+        if v1 in self.vertices and v2 in self.vertices:
+            self.vertices[v1].edges.add(v2)
+            self.vertices[v2].edges.add(v1)
+        else:
+            raise IndexError('That vertex does not exist.')
+    
```

Comment:
I think these should be dft methods if I'm not mistaken. They do look correct though.